File: 0401_0500/0409/0409.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    s = [int(x) for x in f.readline().split()]
    q.write("{:.6f}".format(((s[0] + s[-1]) / 2 + sum(s[1 : n - 1])) / (n - 1)))

File: 0401_0500/0409/0409.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, a, b, m = 0;

	fscanf(f, "%d%d", &n, &a);
	for (int i = 1, k; i < n - 1; ++i)
		fscanf(f, "%d", &k), m += k;
	fscanf(f, "%d",&b);
	fprintf(q, "%f", (m + (double)(a + b) / 2) / (n - 1));
	return 0;
}
